src/app/settings/profile/page.tsx

```diff
@@ -1,7 +1,36 @@
+import { Session } from "next-auth";
+import { getServerSession } from "next-auth/next";
+
+import { ProfileForm } from "@/app/settings/profile/ProfileForm";
+import { Login } from "@/components/login";
+import { authOptions } from "@/lib/auth";
+import { db } from "@/lib/kysely";
+
 export default async function Page() {
+  const session: Session | null = await getServerSession(authOptions);
+  const userId = session?.user?.id;
+
+  if (userId === undefined) {
+    return <Login imgSrc="/images/list-login.png" />;
+  }
+
+  const profile = await db
+    .selectFrom("User")
+    .select(["id", "name", "image", "profileText"])
+    .where("id", "=", userId)
+    .execute();
```

Comment:
[FYI/修正不要]

```suggestion
    .executeTakeFirst();
```
とすると、先頭1件のレコードのみ取得となり返り値が配列でなくなるので(profile2のところ）、後続処理が[0]を書かずに書くことも可能です。
![スクリーンショット 2023-10-01 1 24 47](https://github.com/qin-team-recipe/08-recipe-app/assets/25321850/195bd7a0-73ff-43ec-bd05-e4afbae85098)

参考
https://kysely.dev/docs/getting-started#querying

Reply:
limitとかあるかなって探しはしたんですが、そうやってかけるんですね